### PR TITLE
Fix: PollingFileChangeToken wasn't watching actual files

### PR DESCRIPTION
**Pull request by BrennanConroy**
cc @JunTaoLuo since you showed interest

### CODE REVIEW

**Comment by BrennanConroy** [comment]
@ryanbrandenburg This fixes the flaky tests on macOS

File: test/Microsoft.Extensions.FileProviders.Physical.Tests/PhysicalFileProviderTests.cs [hunk 1]
```diff
@@ -373,34 +377,11 @@ public void TokenIsSameForSamePath()
...
-                            await Task.Delay(WaitTimeForTokenToFire);
-                            Assert.True(token.HasChanged);
-                        }
-                    }
-                }
-            }
-        }
-
-        [Fact]
-        public async Task WatcherWithPolling_ReturnsTrueForChangedFileWhenQueriedMultipleTimes()
```
Comment (BrennanConroy) [inline test/Microsoft.Extensions.FileProviders.Physical.Tests/PhysicalFileProviderTests.cs]: This was a duplicate test, had 0 code differences

**Comment by natemcmaster** [comment]
Do we have a test that consistently fails before this change and is fixed by this? I'm concerned that we didn't catch this sooner, and hope we can add a test that would help catch regressions in this area.

**Comment by BrennanConroy** [comment]
The 3 affected tests in this PR failed on slower machines occasionally. Catching races is difficult.

**Comment by natemcmaster** [comment]
ok, good

File: src/Microsoft.Extensions.FileProviders.Physical/PhysicalFilesWatcher.cs [hunk 2]
```diff
@@ -124,7 +124,7 @@ private IChangeToken GetOrAddFilePathChangeToken(string filePath)
                     new[]
                     {
                         changeToken,
-                        new PollingFileChangeToken(new FileInfo(filePath))
+                        new PollingFileChangeToken(new FileInfo(Path.Combine(_root, filePath)))
```
Comment (natemcmaster) [inline src/Microsoft.Extensions.FileProviders.Physical/PhysicalFilesWatcher.cs]: Should we first check `Path.IsPathRooted` first? Looking at the code, it seems we never validate `filePath` is relative or absolute.

Comment (natemcmaster) [inline src/Microsoft.Extensions.FileProviders.Physical/PhysicalFilesWatcher.cs]: This is the code path I found.
PhysicalFilesWatcher.CreateFileChangeToken(string filter) -> GetOrAddChangeToken(filter) -> GetOrAddFilePathChangeToken(filePath)

Comment (BrennanConroy) [inline src/Microsoft.Extensions.FileProviders.Physical/PhysicalFilesWatcher.cs]: Oh yeah, nice thats a public method

**Comment by BrennanConroy** [comment]
:up: 📅

**Pull request merged**